src/Bindle/BindleClient.cs

```
@@ -15,34 +15,38 @@ namespace Deislabs.Bindle;
 public class BindleClient
 {
     public BindleClient(
-        string baseUri
+        string connectionString
     )
     {
-        _baseUri = new Uri(SlashSafe(baseUri));
-        _httpClient = new HttpClient();
-    }
+        ConnectionInfo connectionInfo = new ConnectionInfo(connectionString);
 
-    public BindleClient(
-        string baseUri,
-        HttpMessageHandler messageHandler
-    )
-    {
-        _baseUri = new Uri(SlashSafe(baseUri));
-        _httpClient = new HttpClient(messageHandler);
+        // TODO: do we want to assume the default listening address?
```

Do we have any defaults for similar uri/url values elsewhere?  First candidate might be `localhost:<port>`... that being said, I think I like err-ing on the side of no defaults and hard error rather than a potentially-hidden default leading to perhaps not as obvious failure mode behavior.

---

Both Hippo and bindle-server default to `http://localhost:8080/v1`:

https://github.com/deislabs/bindle/blob/48839c5ba240b23c1c5bf24f3aaae0c0dac99ce7/bin/server.rs#L38

https://github.com/deislabs/hippo/blob/db8546d938f77960b63d884967ea1cf6b9075e03/src/Infrastructure/Services/BindleService.cs#L16

However given that we're dealing with a connection string I think it's safe to error out if none has been provided.

---

Thanks for the validation! Will remove the TODO